src/Tribe/Repositories/Post_Repository.php

```diff
@@ -0,0 +1,245 @@
+<?php
+/**
+ * Handles all of the post querying.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Tickets\Repositories
+ */
+
+namespace Tribe\Tickets\Repositories;
+
+use Tribe\Tickets\Repositories\Traits\Event;
+use Tribe\Tickets\Repositories\Traits\Post_Attendees;
+use Tribe\Tickets\Repositories\Traits\Post_Tickets;
+use Tribe__Repository;
+use Tribe__Timezones as Timezones;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class Post_Repository.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Tickets\Repositories
+ */
+class Post_Repository extends Tribe__Repository {
+
+	use Post_Attendees;
+	use Post_Tickets;
+	use Event;
+
+	/**
+	 * The unique fragment that will be used to identify this repository filters.
+	 *
+	 * @var string
+	 */
+	protected $filter_name = 'tickets_post';
+
+	/**
+	 * Post_Repository constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->default_args = array_merge( $this->default_args, [
+			'orderby'                      => 'date',
+			'post_status'                  => 'any',
+			'post_type'                    => 'any',
+			// We'll be handling the dates, let's mark the query as a non-filtered one.
+			'tribe_suppress_query_filters' => true,
+		] );
+
+		$this->schema = array_merge( $this->schema, [
+			// These filter methods are added by the Event trait.
+			'ends_after'           => [ $this, 'filter_by_ends_after' ],
+
+			// These filter methods are added by the Post_Tickets trait.
+			'cost'                 => [ $this, 'filter_by_cost' ],
+			'cost_currency_symbol' => [ $this, 'filter_by_cost_currency_symbol' ],
+			'has_tickets'          => [ $this, 'filter_by_has_tickets' ],
+			'has_rsvp'             => [ $this, 'filter_by_has_rsvp' ],
+
+			// These filter methods are added by the Post_Attendees trait.
+			'has_attendees'        => [ $this, 'filter_by_has_attendees' ],
+			'attendee'             => [ $this, 'filter_by_attendee' ],
+			'attendee__not_in'     => [ $this, 'filter_by_attendee_not_in' ],
+			'attendee_user'        => [ $this, 'filter_by_attendee_user' ],
+
+			// This is not yet working, it needs more debugging to determine why it's not functional yet.
+			//'attendee_user__not_in' => [ $this, 'filter_by_attendee_user_not_in' ],
+		] );
+	}
```

Comment:
:no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: c8abab948847d5c33fce913209509ebd</sub>